tests/src/tools/ReadyToRun.SuperIlc/DotnetCli.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace ReadyToRun.SuperIlc
+{
+    /// <summary>
+    /// Helpers to call dotnet CLI via Process.Start
+    /// </summary>
+    static class DotnetCli
+    {
+        // Default 30s timeout for CLI commands
+        private const int DotnetCliTimeout = 30 * 1000;
+
+        public static int New(string workingDirectory, string projectType, StreamWriter logWriter)
+        {
+            return RunProcess("dotnet", $"new {projectType}", workingDirectory, DotnetCliTimeout, logWriter);
+        }
+
+        public static int AddPackage(string workingDirectory, string packageName, StreamWriter logWriter)
+        {
+            return RunProcess("dotnet", $"add package {packageName}", workingDirectory, DotnetCliTimeout, logWriter);
+        }
+
+        public static int Publish(string workingDirectory, StreamWriter logWriter)
+        {
+            return RunProcess("dotnet", "publish", workingDirectory, DotnetCliTimeout, logWriter);
+        }
+
+        private static int RunProcess(string processPath, string arguments, string workingDirectory, int timeout, StreamWriter logWriter)
+        {
+            ProcessStartInfo psi = new ProcessStartInfo()
+            {
+                FileName = processPath,
+                UseShellExecute = false,
+                WorkingDirectory = workingDirectory,
+                Arguments = arguments,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
+            };
+
+            Process p = new Process();
+            p.StartInfo = psi;
+            p.OutputDataReceived += new DataReceivedEventHandler((sender, eventArgs) =>
+            {
+                if (!string.IsNullOrEmpty(eventArgs.Data))
+                {
+                    logWriter.WriteLine(eventArgs.Data);
+                }
+            });
+
+            p.ErrorDataReceived += new DataReceivedEventHandler((sender, eventArgs) =>
+            {
+                if (!string.IsNullOrEmpty(eventArgs.Data))
+                {
+                    logWriter.WriteLine(eventArgs.Data);
+                }
+            });
+
+            p.Start();
+            p.BeginErrorReadLine();
+            p.BeginOutputReadLine();
+
+            if (p.WaitForExit(timeout))
+            {
+                return p.ExitCode;
+            }
+            else
+            {
+                throw new ApplicationException($"Timeout running {processPath} {arguments}");
```

Comment:
Should we try to kill the process if it timed out?

Reply:
Good idea